### PR TITLE
Sequence anyone

### PR DESCRIPTION
**Pull request by JoeCohen**
Any logged in User can add a Sequence to any Observation.

Delivers https://www.pivotaltracker.com/story/show/153851290

Suggested by Steve Russell.  And @AlanRockefeller points out that this is similar to adding a Comment to an Observation -- anyone can do that.

### CODE REVIEW

File: test/controllers/observer_controller_test.rb [hunk 1]
```diff
@@ -927,23 +927,24 @@ def test_show_observation_specimen_stuff
     get(:show_observation, id: obs1.id)
     assert_show_obs(:collection_numbers, [], false)
     assert_show_obs(:herbarium_records, [], false)
-    assert_show_obs(:sequences, [], false)
```
Comment (JoeCohen) [inline test/controllers/observer_controller_test.rb]: Should I move the sequence stuff into a separate test?

**Comment by JoeCohen** [comment]
Forgot about API

**Comment by JoeCohen** [comment]
Updated API.

File: README_API_HELP_MESSAGES.txt [hunk 2]
```diff
@@ -518,7 +518,7 @@ POST sequence
   bases:	string
   locus:	string
   notes:	string
-  observation:	observation (must have edit permission)
```
Comment (JoeCohen) [inline README_API_HELP_MESSAGES.txt]: TRIPLE WOW -- these lines changed automagically! 
Thank you, @pellaea!

**Comment by pellaea** [comment]
Haha, yep, one of the API tests runs the help messages for all the API
endpoints and dumps the result into README_API_HELP_MESSAGES.txt  An awful
lot of redundancy, but at least we know it's up to date! :)

File: app/classes/api/sequence_api.rb [hunk 3]
```diff
@@ -59,8 +59,7 @@ def query_params
 
     def create_params
       {
-        observation: parse(:observation, :observation,
-                           must_have_edit_permission: true),
+        observation: parse(:observation, :observation),
```
Comment (pellaea) [inline app/classes/api/sequence_api.rb]: Oops!  Thanks for catching that!

**Comment by pellaea** [comment]
Looks great.

**Pull request merged**